python/src/equistore/operations/reduce_over_samples.py

```diff
@@ -93,6 +125,22 @@ def _reduce_over_samples_block(
     )
 
     for parameter, gradient in block.gradients():
+        # check if all gradients are zeros
+        if gradient.samples.shape[0] == 0:
+            # The Gradient does not change because, if they are all zeros, also the
+            # gradients of a reduce operation is zero.
+            # For any function of the TensorBlock values x(t):
+            # f(x(t))-> df(x(t))/dx * dx/dt
+            # and dx/dt == 0.
+            for parameter, gradient in block.gradients():
```

Comment:
Why do we have to do a double loop here? You are already looping over the gradients.

Reply:
because of copy-paste